src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -76,8 +79,13 @@ abstract class AbstractCheckTask(
     protected var handledMessageCounter: Long = 0
 
     protected val converter = ProtoToIMessageConverter(VerificationUtil.FACTORY_PROXY, null, null)
-    protected val rootEvent: Event = Event.from(submitTime)
-        .description(description)
+    protected val rootEvent: Event by lazy {
+        Event.from(submitTime)
+                .name(name())
+                .type(type())
+                .bodyData(EventUtils.createMessageBean(eventBody()))
+                .description(description)
+    }
```

Comment:
Why do we need this event to be lazy?

Reply:
If we do just a normal initialization, then we will receive warnings `Calling non-final function name in constructor` in the methods `name()`, `type()`, `eventBody()`. This is the way to avoid it

Reply:
Okay, I get it. But still, it is a hack to avoid warnings. If we will need to get access to the `rootEvent` in the constructor later those methods will be invoked anyway.
I think we can create a `setup` function that received the `Event` as a parameter and the rule implementation can configure the event (we can invoke it here instead of adding abstract method `eventBody()`). The methods for name and type might be useful. But the event's body might contain additional information. So, we should not restrict the ability to configure the root event with this method